src/vanilla/store.ts

```diff
@@ -610,6 +610,7 @@ export const createStore = () => {
 
   const mountAtom = <Value>(
     atom: Atom<Value>,
+    initialListener?: () => void,
     initialDependent?: AnyAtom,
     onMountQueue?: (() => void)[],
```

Comment:
Does this work?
```suggestion
    initialDependent?: AnyAtom,
    onMountQueue?: (() => void)[],
    initialListener?: () => void,
```

Hm, on second thought, this should be a task of `addAtom`. Have you tried it?

Reply:
The problem with pulling it up into `addAtom` is that we need to check whether the value of the atom changed after `onMount`, and if we read the atom's value before we mount its dependents, then the mount order will be changed, and will fail other tests.

Reply:
can we peek the prev value? aka `getAtomState`?

Reply:
If the atom was not read before, then the state would be missing, if I understand it correctly. So we would run the callback even if `onMount` did not change the atom's initial value.

Reply:
okay, I think then this PR isn't what I would do.

I would agree this might not be ergonomic, 
```js
// to keep `value` in sync with the atom
let value = store.get(anAtom)
const cb = () => {
  value = store.get(anAtom)
}
store.sub(anAtom, cb)
// fire it first after the subscription
cb()
```

but, it's in line with our binding:
https://github.com/pmndrs/jotai/blob/bae472500dea29ba876d80b8c066c54816970862/src/react/useAtomValue.ts#L97